src/pages/cart/cart.js

```diff
@@ -1,40 +1,57 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './cart.css';
 import jacket from '../../assets/jacket.jpg';
+import CartCard from '../../components/cartCard/cartCard';
+import { useAuth } from '../../context/authContext';
+import axios from 'axios';
+import { useNavigate } from 'react-router';
 
 export default function Cart() {
+    const { user, encodedToken } = useAuth();
+    const [cartdata, setCartdata] = useState([]);
+    const navigate = useNavigate();
+    useEffect(async () => {
+        if (user !== null) {
+            const res = await axios.get("/api/user/cart", {
+                headers: {
+                    authorization: encodedToken
+                }
+            });
+            console.log(res.data.cart);
+            setCartdata(res.data.cart);
+        }
+        else {
+            console.log("Navigating to login page.");
+            navigate("/login");
+            alert("Please login first to see your cart.");
+        }
+
+    }, [user])
```

Comment:
SUGGESTION : A better way to do this would be to have a cart context and move this functionality there.